homeassistant/components/bluesound/media_player.py

```diff
@@ -449,6 +447,22 @@ async def async_update_status(self):
             _LOGGER.info("Client connection error, marking %s as offline", self._name)
             raise
 
+    @property
+    def unique_id(self):
+        """Return an unique ID."""
+        return f"{format_mac(self._sync_status.get('@mac'))}-{self.port}"
+
+    @property
+    def device_info(self):
+        """Return a device description for device registry."""
+        return {
+            "identifiers": {(DOMAIN, self.unique_id)},
+            "name": self._sync_status.get("@name"),
+            "manufacturer": self._sync_status.get("@brand"),
+            "model": f"{self._sync_status.get('@modelName')} {self._sync_status.get('@model')}",
+            "sw_version": self._sync_status.get("@schemaVersion"),
+        }
```

Comment:
```suggestion
        return DeviceInfo(
            identifiers={(DOMAIN, self.unique_id)},
            name=self._sync_status.get("@name"),
            manufacturer=self._sync_status.get("@brand"),
            model=f"{self._sync_status.get('@modelName')} {self._sync_status.get('@model')}",
            sw_version=self._sync_status.get("@schemaVersion"),
        }
```

Please use the typed dataclass.

`from homeassistant.helpers.entity import DeviceInfo`

Reply:
Thanks, the doc seems to be outdated: https://developers.home-assistant.io/docs/device_registry_index#automatic-registration-through-an-entity